addons/auth_signup/controllers/main.py

```diff
@@ -54,6 +55,11 @@ def web_auth_signup(self, *args, **kw):
                     _logger.error("%s", e)
                     qcontext['error'] = _("Could not create a new account.")
 
+        elif 'p_id' in qcontext:
+            partner = request.env['res.partner'].sudo().browse(int(qcontext.get('p_id')))
+            if partner and partner.user_ids.state != 'new':
```

Comment:
That's going to break if the partner has more than one user. Also checking if the user(s) are `"active"` means you don't even have to check if the partner exists.

Which incidentally it always does, because you created a recordset and the existence is *not validated by default*. So by definition `partner` is always a non-empty recordset, and thus truthy.